app/controllers/rents_controller.rb

```diff
@@ -28,48 +29,30 @@ def edit
       @profile = @rent.profile
       @utskott = @rent.profile.car_councils
     end
-
   end
 
   def create
     @rent = Rent.new_with_status(rent_params, current_user)
-    respond_to do |format|
-      if @rent.save
-        if (current_user)
-          format.html { redirect_to @rent, notice: 'Bokningen skapades!' }
-        else
-          format.html { redirect_to @rent, notice: 'Bokningen skapades, du ska ha fått ett mejl till angiven epostadress (möjligen i skräppost)!' }
-        end
-        format.json { render json: @rent, status: :created, location: @rent }
-      else
-        format.html { render action: "new" }
-        format.json { render json: @rent.errors, status:  :unprocessable_entity }
-      end
-    end
+    flash[:notice] = "Bokningen skapades, skickat mailbekräftelse." if @rent.save
+    respond_with @rent
   end
 
   def update
-    respond_to do |format|
-      if @rent.update_with_authorization(rent_params,current_user)
-        format.html { redirect_to edit_rent_path(@rent), notice: 'Bokningen uppdaterades.' }
-        format.json { render json: @rent, status: :created, location:  @rent }
-      else
-        format.html { render action: "edit"}
-        format.json { render json: @rent.errors, status:  :unprocessable_entity }
-      end
+    if @rent.update_with_authorization(rent_params, current_user)
+      flash[:notice] = 'Bokningen uppdaterades'
+      redirect_to edit_rent_path(@rent)
+    else
+      render action: :edit
     end
   end
 
   def destroy
     @rent.destroy
-    respond_to do |format|
-      format.html { redirect_to :bil, notice: 'Bokningen raderades.' }
-      format.json { head :no_content }
-    end
+    flash[:notice] = "Bokningen raderades"
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.